tests/mainnet/test_getters.py

```diff
@@ -0,0 +1,726 @@
+import itertools
+import warnings
+
+import brownie
+import pytest
+
+from .abis import curve_pool, curve_pool_v2, gauge_controller
+from .utils.constants import (
+    METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX,
+    METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX,
+    METAREGISTRY_STABLE_FACTORY_HANDLER_INDEX,
+    METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX,
+)
+
+MAX_REGISTRIES = 4
+MAX_POOLS_TO_CHECK = 200
```

Comment:
Why would you make this a constant? You have a fixture for the pool count of every registry from which you could get the total number of pools.

Reply:
look at using lazy fixtures, but see comments below, imo a loop would also have the advantage of being faster

Reply:
Going with loop. lazy fixtures is borking for me.